core/src/main/java/org/apache/iceberg/deletes/SortingPositionOnlyDeleteWriter.java

```diff
@@ -123,22 +139,35 @@ private DeleteWriteResult writeDeletes(Collection<CharSequence> paths) throws IO
     }
 
     FileWriter<PositionDelete<T>, DeleteWriteResult> writer = writers.get();
+    List<DeleteFile> rewrittenDeleteFile = Lists.newArrayList();
 
     try {
       PositionDelete<T> positionDelete = PositionDelete.create();
       for (CharSequence path : sort(paths)) {
```

Comment:
Another aspect I'm curious about, have we ever compared with using a `TreeMap` instead of sorting? It'll be the same time complexity in the end but interested in seeing if there's any significant differences in practice.

Reply:
I'd say `TreeMap` starts to make sense if we access the collection in sorted order more than once. Otherwise, paying the extra cost during inserts may not be worth it.